Automattic/jetpack
# Dialogue: update media-player-control Full version

This is a suggestion that empowers the MediaPlayerControl to provide additional options to deal with the timestamp value and its synchronization with the media file. 
This is an alternative to the approach suggested here https://github.com/Automattic/jetpack/pull/18390

<img src="https://user-images.githubusercontent.com/77539/105229683-1351da80-5b43-11eb-8c4f-24aac9ad30a7.png" width="700px" />


### Block toolbar
In order to make the UX simpler and easy to use, we've just kept the play/pause button and the current time display there.

<img width="500" alt="Screen Shot 2021-01-20 at 5 11 12 PM" src="https://user-images.githubusercontent.com/77539/105229337-8f97ee00-5b42-11eb-9b82-99dc9fe8276e.png">

Either the play/pause button, as well as the current-time display, are tied to the podcast player globally. Think of this as a shorthand of the global player.

### Block settings (Sidebar)

When the MediaPlayerControl is rendered in the block sidebar context, it changes considerably.

<img width="500" src="https://user-images.githubusercontent.com/77539/105229933-6e83cd00-5b43-11eb-9441-33ef6975742f.png" />

The most remarkable point here is the **control is tied to the timestamp value, as default**.
It means if the user clicks on the `skip forward` or `jump back` button, it will change the **timestamp** value, while the podcast player will keep the same state.

It's possible to connect the Media Player Control with the Podcast player and vice versa, enabling the Keep in-sync option by clicking on the respective button. If so, it will keep the components connected which is very powerful when the user needs to deal with the synchronization between a bunch of dialogue blocks and the audio file.

Keep in-sync **enabled** | Keep in-sync **disabled**
---------|---------- 
![image](https://user-images.githubusercontent.com/77539/105231098-164dca80-5b45-11eb-8c14-7df5556b275c.png) | ![image](https://user-images.githubusercontent.com/77539/105231143-2f567b80-5b45-11eb-9bfe-752549abeeb9.png)

In the next example, the user syncs the dialogue with the audio file using this mode. After deactivating the keep in-sync mode, it checks the timestamp by clicking on the `Play in custom time` button.

https://user-images.githubusercontent.com/77539/105231894-56617d00-5b46-11eb-9c00-fd63365731e4.mov

Fixes #

#### Changes proposed in this Pull Request:

* Dialogue: update media-player-control Full version

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






* Add a Podcast player to the editor
* Add a Conversation block
* Set custom timestamp for each Dialogue block
* Check all features described above

#### Proposed changelog entry for your changes:


*
